pkg/sparrow/run.go

```diff
@@ -135,49 +136,46 @@ func (s *Sparrow) Run(ctx context.Context) error {
 	}
 }
 
-// ReconcileChecks registers new Checks, unregisters removed Checks,
-// resets the Configs of Checks and starts running the checks
+// ReconcileChecks reconciles the checks.
+// It registers new checks, updates existing checks and unregisters checks not in the new config.
 func (s *Sparrow) ReconcileChecks(ctx context.Context, cfg runtime.Config) {
```

Comment:
I'm angry at myself, because this function should be a part of the Controller you wrote. And it currently can't be because of the targetManager.

Reply:
We can if we move the target enrichment out of the `ReconcileChecks` method.

Reply:
I think that the check controller should have a Run method as well, as do all other components. It should just react to the new `runtime.Config` instances coming in via the channel and register/update/delete checks from the state.

I'm excited to see how you'll separate the enrichTargets from the sparrow.

Reply:
I was taking the channel route again (a channel for the `globalTargets` read from the `targetManager`), as one way of decoupling the `targetManager` from the sparrow and making the targets available to those who need it.

Reply:
I'd rather not implement the `Run` method as long as your target manager channel suggestion isn't implemented.

A `Run` method currently would complicate the `CheckController` with adding an unnecessary channel, because both, the sparrow, and the controller would need the `cRuntime` channel. The sparrow would then send the enriched config through the channel to the controller. 

But if we'll refactor the behavior of the target manager, the `Run` method should be implemented.